iotdb-core/datanode/src/test/java/org/apache/iotdb/db/queryengine/plan/planner/LocalExecutionPlannerTest.java

```diff
@@ -0,0 +1,200 @@
+              pipelineDriverFactory ->
+                  Assert.assertEquals(
+                      MOCK_SERIES_SCAN.calculateMaxPeekMemory(),
+                      pipelineDriverFactory.getEstimatedMemorySize()));
+
+      context.addPipelineDriverFactory(
+          root, context.getDriverContext(), root.calculateMaxPeekMemory());
+
+      long expected =
+          4 * MOCK_SERIES_SCAN.calculateRetainedSizeAfterCallingNext()
+              + context.getPipelineDriverFactories().stream()
+                      .map(PipelineDriverFactory::getEstimatedMemorySize)
+                      .reduce(0L, Long::sum)
+                  / 5
+                  * Math.max((QUERY_THREAD_COUNT / ESTIMATED_FI_NUM), 1);
+
+      Assert.assertEquals(
+          expected, calculateEstimatedMemorySize(context.getPipelineDriverFactories()));
+    } catch (Exception e) {
+      Assert.fail();
+    }
+  }
+
+  /**
+   * This test will test dop = 2. Expected result is two pipelines:
+   *
+   * <p>The first is: TimeJoin1 - [SeriesScan1, SeriesScan0, ExchangeOperator];
+   *
+   * <p>The second is: ExchangeOperator - TimeJoin1-1[SeriesScan2, SeriesScan3].
+   */
+  @Test
+  public void testCheckMemoryWithTimeJoin() throws IllegalPathException {
+    TypeProvider typeProvider = new TypeProvider();
+    FullOuterTimeJoinNode timeJoinNode = initFullOuterTimeJoinNode(typeProvider, 4);
+    LocalExecutionPlanContext context = createLocalExecutionPlanContext(typeProvider);
+    context.setDegreeOfParallelism(2);
+
+    Operator root = timeJoinNode.accept(new OperatorTreeGenerator(), context);
+
+    context.addPipelineDriverFactory(
+        root, context.getDriverContext(), root.calculateMaxPeekMemory());
+
+    long expected =
+        context
+                .getPipelineDriverFactories()
+                .get(0)
+                .getOperation()
+                .calculateRetainedSizeAfterCallingNext()
+            + context
+                .getPipelineDriverFactories()
+                .get(1)
+                .getOperation()
+                .calculateRetainedSizeAfterCallingNext()
+            + context.getPipelineDriverFactories().stream()
+                    .map(PipelineDriverFactory::getEstimatedMemorySize)
+                    .reduce(0L, Long::sum)
+                / 2
+                * Math.max((QUERY_THREAD_COUNT / ESTIMATED_FI_NUM), 1);
+
+    Assert.assertEquals(
+        expected, calculateEstimatedMemorySize(context.getPipelineDriverFactories()));
+  }
+
+  private long calculateEstimatedMemorySize(
+      final List<PipelineDriverFactory> pipelineDriverFactories) {
+    long result = 0;
+    try {
+      LocalExecutionPlanner localExecutionPlanner = new LocalExecutionPlanner();
+      long res = localExecutionPlanner.calculateEstimatedMemorySize(pipelineDriverFactories);
+      Method method =
+          LocalExecutionPlanner.class.getDeclaredMethod("calculateEstimatedMemorySize", List.class);
+      method.setAccessible(true); // 设置方法为可访问
```

Comment:
Wow, better use English for comments, m3 bro?

Reply:
You are right, copilot sucks